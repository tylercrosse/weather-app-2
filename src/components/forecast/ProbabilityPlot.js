/* eslint-disable array-callback-return */

import React from 'react';
import PropTypes from 'prop-types';
import {
  XYPlot,
  YAxis,
  HorizontalGridLines,
  VerticalGridLines,
  AreaSeries,
  LineSeries,
  Crosshair
} from 'react-vis';

/**
 * Format items for Crosshair.
 * @param {Array} values Array of values.
 * @returns {*} Formatted list of items.
 */
function itemsFormat(values) {
  if (values) {
    return [
      { value: `${Math.round(values[0].y * 100)}%`, title: 'cloud cover' },
      { value: `${Math.round(values[1].y * 100)}%`, title: 'chance of rain' },
      { value: `${Math.round(values[2].y * 100)}%`, title: 'humidity' }
    ];
  }
}

const ProbabilityPlot = props => {
  return (
    <XYPlot
      animation
      className="forecast__chart chart__prob"
      yDomain={[0, 1]}
      height={160}
      width={props.chartWidth}
      onMouseLeave={props.handleMouseLeave}
    >
      <YAxis hideLine left={6} tickFormat={v => `${v * 100}%`} />
      <HorizontalGridLines />
      <VerticalGridLines tickValues={props.dayDivsions} />
      <LineSeries
        color="grey"
        opacity={0.4}
        data={[{ x: props.currentTime, y: 0 }, { x: props.currentTime, y: 1 }]}
      />
      <AreaSeries
        data={props.probNightData}
        style={{
          stroke: 'none',
          fill: 'rgba(130, 130, 130, 0.1)'
        }}
      />
      <AreaSeries
        color="rgb(25, 35, 250)"
        curve="curveMonotoneX"
        data={props.percipProbData}
        style={{
          fill: 'rgba(25, 35, 250, 0.5)'
        }}
      />
      <AreaSeries
        color="grey"
        curve="curveMonotoneX"
        data={props.cloudCoverData}
        style={{
          fill: 'rgba(130, 130, 130, 0.2)'
        }}
        onNearestX={props.handleNearestX}
      />
      <LineSeries
        color="#2BEEA3"
        curve="curveMonotoneX"
        data={props.humidityData}
        style={{
          fill: 'none'
        }}
      />
      <Crosshair
        values={props.crosshairValues}
        titleFormat={props.titleFormat}
        itemsFormat={itemsFormat}
      />
    </XYPlot>
  );
};

ProbabilityPlot.propTypes = {
  chartWidth: PropTypes.number.isRequired,
  crosshairValues: PropTypes.array.isRequired,
  handleMouseLeave: PropTypes.func.isRequired,
  handleNearestX: PropTypes.func.isRequired,
  dayDivsions: PropTypes.array.isRequired,
  titleFormat: PropTypes.func.isRequired,
  currentTime: PropTypes.number.isRequired,
  probNightData: PropTypes.array.isRequired,
  percipProbData: PropTypes.array.isRequired,
  humidityData: PropTypes.array.isRequired,
  cloudCoverData: PropTypes.array.isRequired,
}

export default ProbabilityPlot;
