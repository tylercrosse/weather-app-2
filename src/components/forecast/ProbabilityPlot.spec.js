import React from 'react';
import { shallow } from 'enzyme';
import toJson from "enzyme-to-json";
import ProbabilityPlot from './ProbabilityPlot';

const setup = () => {
  const props = {
    chartWidth: 880,
    crosshairValues: [],
    handleMouseLeave: jest.fn(),
    handleNearestX: jest.fn(),
    dayDivsions: [],
    titleFormat: jest.fn(),
    currentTime: 1499105000,
    probNightData: [],
    percipProbData: [],
    humidityData: [],
    cloudCoverData: [],
  }

  const component = <ProbabilityPlot {...props} />;
  const wrapper = shallow(component);

  return {
    props,
    wrapper
  }
}

describe('<ProbabilityPlot />', () => {
  it('should render correctly', () => {
    const { wrapper } = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
})
