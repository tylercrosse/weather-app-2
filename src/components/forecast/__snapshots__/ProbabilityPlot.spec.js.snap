// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProbabilityPlot /> should render correctly 1`] = `
<XYPlot
  animation={true}
  className="forecast__chart chart__prob"
  height={160}
  onMouseLeave={[Function]}
  width={880}
  yDomain={
    Array [
      0,
      1,
    ]
  }
>
  <YAxis
    attr="y"
    hideLine={true}
    left={6}
    orientation="left"
    tickFormat={[Function]}
  />
  <HorizontalGridLines
    attr="y"
    direction="horizontal"
  />
  <VerticalGridLines
    attr="x"
    direction="vertical"
    tickValues={Array []}
  />
  <LineSeries
    color="grey"
    curve={null}
    data={
      Array [
        Object {
          "x": 1499105000,
          "y": 0,
        },
        Object {
          "x": 1499105000,
          "y": 1,
        },
      ]
    }
    opacity={0.4}
    strokeStyle="solid"
    style={Object {}}
  />
  <AreaSeries
    className=""
    data={Array []}
    style={
      Object {
        "fill": "rgba(130, 130, 130, 0.1)",
        "stroke": "none",
      }
    }
  />
  <AreaSeries
    className=""
    color="rgb(25, 35, 250)"
    curve="curveMonotoneX"
    data={Array []}
    style={
      Object {
        "fill": "rgba(25, 35, 250, 0.5)",
      }
    }
  />
  <AreaSeries
    className=""
    color="grey"
    curve="curveMonotoneX"
    data={Array []}
    onNearestX={[Function]}
    style={
      Object {
        "fill": "rgba(130, 130, 130, 0.2)",
      }
    }
  />
  <LineSeries
    color="#2BEEA3"
    curve="curveMonotoneX"
    data={Array []}
    opacity={1}
    strokeStyle="solid"
    style={
      Object {
        "fill": "none",
      }
    }
  />
  <Crosshair
    itemsFormat={[Function]}
    titleFormat={[Function]}
    values={Array []}
  />
</XYPlot>
`;
